pnet-idp-client-saml2/src/main/java/at/porscheinformatik/idp/saml2/response/CheckRelayStateMessageHandler.java

```diff
@@ -0,0 +1,31 @@
+package at.porscheinformatik.idp.saml2.response;
+
+import org.apache.commons.lang3.ObjectUtils;
+import org.opensaml.messaging.context.MessageContext;
+import org.opensaml.messaging.handler.MessageHandlerException;
+
+import static at.porscheinformatik.idp.saml2.Saml2ResponseProcessor.Saml2AuthenticationTokenContext;
+
+public class CheckRelayStateMessageHandler extends AbstractSimpleMessageHandler
+{
+
+    @Override
+    public void invoke(MessageContext messageContext) throws MessageHandlerException
+    {
+        Saml2AuthenticationTokenContext authenticationContext =
+            messageContext.getSubcontext(Saml2AuthenticationTokenContext.class);
+        String responseRelayState = authenticationContext
+            .getResponseRelayState()
+            .orElseThrow(() -> new MessageHandlerException("Response is missing relay state."));
+        String requestedRelayState = authenticationContext
+            .getRequestedRelayState()
+            .orElseThrow(
+                () -> new MessageHandlerException("Requested relay state doesn't match relay state in response"));
+
+        if (ObjectUtils.notEqual(responseRelayState, requestedRelayState))
```

Comment:
I would prefer not to depend on commons lang, only for a comparison here. Please change this line to `!Objects.equals(..., ...)` instead.

Reply:
Change like requested!